cassandra/cluster.py

```diff
@@ -2262,21 +2281,22 @@ def prepare(self, query, custom_payload=None):
         if self.cluster.prepare_on_all_hosts:
             host = future._current_host
             try:
-                self.prepare_on_all_hosts(prepared_statement.query_string, host)
+                self.prepare_on_all_hosts(prepared_statement.query_string, host, prepared_keyspace)
             except Exception:
                 log.exception("Error preparing query on all hosts:")
 
         return prepared_statement
 
-    def prepare_on_all_hosts(self, query, excluded_host):
+    def prepare_on_all_hosts(self, query, excluded_host, keyspace=None):
```

Comment:
Same as with prepare, should keyspace be added here as a parameter? I'm more inclined to add it here but maybe the best is to add it together with `session.prepare` since it's only called from inside that function.